config/config.go

```diff
@@ -7,19 +7,40 @@ import (
 
 // Config holds the application configuration
 type Config struct {
-	DiceAddr      string
-	ServerPort    string
-	RequestLimit  int // Field for the request limit
-	RequestWindow int // Field for the time window in seconds
+	DiceDB struct {
+		Addr     string // Field for the Dice address
+		Username string // Field for the username
+		Password string // Field for the password
+	}
+	Server struct {
+		Port          string // Field for the server port
+		RequestLimit  int    // Field for the request limit
+		RequestWindow int    // Field for the time window in seconds
+	}
 }
 
 // LoadConfig loads the application configuration from environment variables or defaults
 func LoadConfig() *Config {
 	return &Config{
-		DiceAddr:      getEnv("DICE_ADDR", "localhost:7379"), // Default Dice address
-		ServerPort:    getEnv("SERVER_PORT", ":8080"),        // Default server port
-		RequestLimit:  getEnvInt("REQUEST_LIMIT", 1000),      // Default request limit
-		RequestWindow: getEnvInt("REQUEST_WINDOW", 60),       // Default request window in seconds
+		// if host and port is combined then it is called
```

Comment:
i dont think the naming conventions are clashing Currently. we can have a flat Config file, rather than adding structs.